sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -8887,6 +8979,374 @@ prefetch(const T *p, PropertyListT props = {}) {
+/// each address. The parameter 'N' must be divisible by 'VS'.
+/// @param acc Accessor referencing the data to load.
+/// @param byte_offsets the vector of offsets in bytes. If force stateless
+/// memory is used the offsets can be up to 64 bit size, otherwise up to 32 bit
+/// size. For each i, (acc + byte_offsets[i]) must be element size aligned.
+/// @param props The optional compile-time properties. Only cache hint
+/// properties are used.
+template <typename T, int N, int VS = 1, typename OffsetSimdViewT,
+          typename AccessorT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    detail::is_device_accessor_with_v<AccessorT,
+                                      detail::accessor_mode_cap::can_read> &&
+    detail::is_simd_view_type_v<OffsetSimdViewT> &&
+    ext::oneapi::experimental::is_property_list_v<PropertyListT>>
+prefetch(AccessorT acc, OffsetSimdViewT byte_offsets,
+         PropertyListT props = {}) {
+  prefetch<T, N, VS>(acc, byte_offsets.read(), props);
+}
+
+/// template <typename T, int VS = 1, typename AccessorT, typename OffsetT,
+///           typename PropertyListT = empty_properties_t>
+/// void prefetch(AccessorT acc, uint32_t byte_offset, simd_mask<1> mask,
+///                   PropertyListT props = {});                   // (acc-pf-7)
+/// Supported platforms: DG2, PVC only.
+/// Prefetches elements of the type 'T' from continuous memory location
+/// addressed by the accessor \p acc, and offset \p byte_offset and the length
+/// \p VS elements into the cache.
+/// @tparam T Element type.
+/// @tparam VS Vector size. It specifies the number of consequent elements to
+/// prefetch.
+/// @param acc Accessor referencing the data to load.
+/// @param byte_offset offset from the base address.
+/// @param mask The access mask. If it is set to 0, then the prefetch is
+/// omitted.
+/// @param props The optional compile-time properties. Only cache hint
+/// properties are used.
+template <typename T, int VS = 1, typename AccessorT, typename OffsetT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    std::is_integral_v<OffsetT> &&
+    detail::is_device_accessor_with_v<AccessorT,
+                                      detail::accessor_mode_cap::can_read> &&
+    ext::oneapi::experimental::is_property_list_v<PropertyListT>>
+prefetch(AccessorT acc, OffsetT byte_offset, simd_mask<1> mask,
+         PropertyListT props = {}) {
+#ifdef __ESIMD_FORCE_STATELESS_MEM
+  prefetch<T, VS>(detail::accessorToPointer<T>(acc), byte_offset, mask, props);
+#else
+  constexpr auto L1Hint =
+      detail::getPropertyValue<PropertyListT, cache_hint_L1_key>(
+          cache_hint::uncached);
+  constexpr auto L2Hint =
+      detail::getPropertyValue<PropertyListT, cache_hint_L2_key>(
+          cache_hint::cached);
```

Comment:
@fineg74 - I did not notice (during the main code-review) that the default values for {L1, L2} are {uncached, cached}, e.g. not `none` as for all other functions.
Why were those default values used?

Reply:
Using {uncached, cached} as default is not obvious. I see at least 2 alternatives here:
1) If there is some good reason for using {uncached, cached}, it should be documented properly. 
2) Set the defaults to {none, none}, then depending on GPU driver and GenX prefetch intrinsic... if it accepts `none`, then allow `none`, otherwise, static_assert inside check_cache_hints().